### DISCUSSION TITLE
Is it possible to disable input button for X amount of seconds after pressing.

### DISCUSSION THREAD

**Discussion by james-dmu** [Lillifee/raspiCam#50] [Ideas]
Hi,

Firstly great work I am finding raspiCam works very well for a newly build IR camera however I have discovered an issue (mostly with the way I would like to use the software) to do with using a PIR to trigger capture. This works ok for images however I am finding the PIR will also randomly trigger to stop the video recording due to repeat signals from the GPIO pin. The GPIO trigger is set to "rising only" however if the GPIO input goes low and then rises again during the video it will stop the recording. I would like a way (if possible) to disable input from the GPIO for a specified duration. That way it would continue to record and only restart another recording after that period has passed. So say I tell it to do a 1 minute recording and wait 1 minute 5 before listening to the GPIO pin again.

I could adjust the PIR boards potentiometer for the delay however this then would have an adverse effect on photo captures as it would then wait 1 minute or so too shoot another image and I ideally want photo after photo as soon as it can shoot.

In short I want to ideally be able to trigger a X amount of seconds video and it not stop for that duration but also allow in photo mode it to just retake ASAP.

Hope that makes sense and again the software is really impressive so keep up the good work.

Thanks.

James

**Comment by Lillifee** [comment]
Hello @james-dmu,

sorry for the late responce. I have made some improvements to the GPIO settings. One of the options is a lockout time that disables the GPIO trigger for a specific period. The second option is to stop the capture on trigger. If deactivated, it will no longer stop the capture if it is already running, which should resolve your issue as well.

Unfortunately, I haven't had the opportunity to test these changes as I don't have a button wired up at the moment. However, I will create a new release once I've had the chance to thoroughly test it.

Thank you for your patience and understanding.

[raspiCam.tar.gz](https://github.com/Lillifee/raspiCam/files/11370628/raspiCam.tar.gz)

**Reply by james-dmu** [reply]
That sounds great. I'll look forward to it trying it once you get it sorted. It sounds like those changes will work perfectly for what I need with the PIR so thank you for your hard work on this.

The app works really well at the moment and this will be the icing on the cake for my use. :)

James

**Reply by Lillifee** [reply]
Hey @james-dmu,

I have tested the release yesterday evening and everything appeared to be functioning properly. 
I created a new release, which you can find at https://github.com/Lillifee/raspiCam/releases/tag/v2.0.6. 
Please let me know if it works for you.

Additionally, I am interested in hearing more about how you use raspicam. 
If you have some time to explain or share some photos, I would greatly appreciate it. You can find the discussion thread at https://github.com/Lillifee/raspiCam/discussions/28. 

Thank you!

**Reply by james-dmu** [reply]
Thanks for adding those features so quickly. I'll try to test it out as soon as I get chance in the next few days and let you know how it goes. I'll also sort out an explanation etc of how I am using it. :)

James

**Reply by james-dmu** [reply]
I tested the new release tonight and it seems to work perfectly so thanks again for sorting that so quickly. I've been able to set my PIR to its minimum delay of 3 seconds and so it keep taking photos in photo mode when there is movement and also in video with the GPIO lockout I can leave the PIR set at the same 3 seconds but tell RaspiCam to start a 1 minute video and then to ignore the GPIO for 1 minute 3 seconds (1 minute may well work too but I tested with a slightly longer time).

I haven't tested the other button yet which can control if to stop the capture on another button press or not but will do soon as both will prove very useful to me I think.

I'll get some sort of write up and some images done in the next few days hopefully for you.

Thanks again, it's really appreciated.
